Blog post introducing aiapy

wtbarnes opened the pull request:
This PR adds a blog post entry introducing the aiapy package and demos a few components of the package. This is meant to coincide (roughly) with the sunpy 2.0 release and the deprecation of `sunpy.instr.aia.aiaprep`.

**This is currently a work in progress and only a rough outline at this point.**

**This PR needs to be blocked until the release of aiapy v0.2 because of an incompatibility with how sunpy v2.0 deals with submaps. In v0.2, this has been fixed and the sunpy dependency is pinned to >=2.0**

Fixes https://github.com/sunpy/sunpy.org/issues/226

wtbarnes commented:
ping @fluxtransport

wtbarnes commented:
v0.2 is now out so I really need to get on this 😅

drsophiemurray commented:
> v0.2 is now out so I really need to get on this 😅

Congrats on 0.2 🎉  That would be great if you can work on this soon! @hayesla will be writing a more general affiliated package this week so could you plan to have yours ready a week or so after thats out? Then I'll move onto my next victims 😈

nabobalis commented:
I guess this will need updating for aiapy 0.3.0?

wtbarnes commented:
Yep. I have a much more complete draft that I just need to push.

wtbarnes commented:
Ok so this has dragged on a really long time and all due to my laziness 😅. I've just updated the notebook and I think all of the content is there. 

I've organized this by querying two AIA images at the same time stamp and then taking them through every stage of analysis, starting with respiking and ending with exposure time normalization. Opinions on this?

Just a few things left to do:

- [ ] PSF example. Do we want to include this? The only reason I haven't done this yet is because I need to hop onto another machine to use the GPU.
- [ ] Explanatory text. There needs to be text explaining each of the sections and what's going on in each section. A lot of this can be copied from the docs.
- [ ] Copy edit. This could use one more pass through to make sure everything is coherent. 

A few questions:

* We're close to a v0.4. Do we want to wait on that before publishing this?
* The aiapy JOSS paper is under review.  Should we wait on that as well?

nabobalis commented:
>Few questions:
> 
> * We're close to a v0.4. Do we want to wait on that before publishing this?
> * The aiapy JOSS paper is under review.  Should we wait on that as well?
Why not?

nabobalis commented:
I've rebased this and updated some parts. It needs a review.

Cadair commented:
![image](https://user-images.githubusercontent.com/1391051/139832486-6ae96714-9439-4882-9ac3-9658355b7347.png)

rst intersphinx links don't work in markdown :wink:

nabobalis commented:
> ![image](https://user-images.githubusercontent.com/1391051/139832486-6ae96714-9439-4882-9ac3-9658355b7347.png)
> 
> rst intersphinx links don't work in markdown 😉

Thats ok.

Cadair commented:
no?

nabobalis commented:
I've cleaned out the cell output, so you can open the diff and do code suggestions or text improvements.

nabobalis commented:
So it will run the notebook if the cell outputs are empty.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Query 2-3 AIA images from different wavelengths and create [`sunpy.map.Map`](https://docs.sunpy.org/en/stable/code_ref/map.html) objects."
```
why is it 2-3? should this just say 2 images?

wtbarnes replied:
This wording has been removed and more exlanatory text added.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you want to use lots of AIA images and compare them to each other, you will need to create level 1.5 data.\n",
```
I think a better sentence should be included here rather than "if you want to use lots of AIA images". 

```suggestion
    "With aiapy, you can create level 1.5 data from the level 1.0 data provided in the FITS files that we get from the data provider. To create to level 1.5 data, the AIA map needs to have its pointing updated, the roll angle removed and rescaled to a common plate scale. Converting a level 1 to level 1.5 is often known as `aia_prep`.",
```

wtbarnes replied:
I've reworded the text in this section to be a bit more clear. Let me know if you think it still needs a bit.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Respiking an AIA image is quite straightfoward."
```
maybe the example gallery of each of these examples could be added here - just for example someone wants to know more about respiking etc

wtbarnes replied:
I've now included more information about the respiking procedure and linked to the example.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   "source": [
+    "As all things do in life do, they age, fail and deteriorate until they die.\n",
+    "\n",
+    "While we can't fix death, we can fix AIA filter degradation."
```
😆

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+    "As all things do in life do, they age, fail and deteriorate until they die.\n",
+    "\n",
+    "While we can't fix death, we can fix AIA filter degradation."
+   ]
```
Could another sentence be added here about the actual degradation of the channels - even just pulling straight from the aiapy example: "The performance of the AIA telescope is unfortunately degrading over time, leading to the resulting images becoming increasingly dim."

wtbarnes replied:
I've now just copied the introductory text from the example and also added the equation from the docstring.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "t_begin = astropy.time.Time('2010-03-25T00:00:00')\n",
```
is there a reason this is with astropy time rather than parse_time?

wtbarnes replied:
I've switched to `parse_time` everywhere now. Not sure why I wasn't using it before.

hayesla commented on posts/2021/2021-10-01-aiapy-demo.ipynb:
```diff
@@ -0,0 +1,569 @@
+   ]
+  },
+  {
+   "cell_type": "code",
```
could it also be shown here that the `bunit` and `exposure_time` on the map has been updated?

wtbarnes replied:
Done!

hayesla reviewed (approved):
This looks great! 

I think it would be worthwhile to include a little bit more of an introductory paragraph/few sentances about AIApy and why its needed? just even pulled from the aiapy docs or the JOSS paper?

wtbarnes commented:
Ok I've made some good progress on this, but still have a few sections to go. I'm aiming to finish this by tomorrow. I'm trying to address Laura's comments as I go. I will also try to ping Mark once I'm done so he can look over it.

- [ ] Degradation section
- [ ] Uncertainty section (I added this)
- [ ] Exposure time section
- [ ] Wavelength response section

fluxtransport commented:
Thanks for the updates and comments by @wtbarnes, @hayesla @nabobalis. It looks pretty good. Minor comments from me only. 
- Should the VSO query example be followed up with a paragraph about the limitations of using the VSO for SDO data queries? E.g. that preceded the JSOC example links. 
- The side by side comparison of original vs deconvolved image doesn't show much of a difference. If it's not going to wreak havoc on the rest of the examples, perhaps choose a different image where the PSF is more pronounced? Alternatively, show a line plot across a dark feature to highlight the difference?
- The Corrected vs Uncorrected images (for degradation) also look to similar. Perhaps choose a date later in the mission?

wtbarnes commented:
> * Should the VSO query example be followed up with a paragraph about the limitations of using the VSO for SDO data queries? E.g. that preceded the JSOC example links.

I added an extra sentence to this effect. If you have additional text you'd like to add, I can paste that in too.

> * The side by side comparison of original vs deconvolved image doesn't show much of a difference. If it's not going to wreak havoc on the rest of the examples, perhaps choose a different image where the PSF is more pronounced? Alternatively, show a line plot across a dark feature to highlight the difference?

I changed to using an image just after the Sept 10 flare and created the subplot around the post-flare loops. This emphasizes the removal of the diffraction pattern around the bright loop top. I also added a line plot across the loop top as a function of the helioprojective longitude.

> * The Corrected vs Uncorrected images (for degradation) also look to similar. Perhaps choose a date later in the mission?

Done. I'm using a 335 image from 9/10/2017. The difference is much more obvious now.

nabobalis commented:
Thanks everyone!
